### DISCUSSION TITLE
TeslaMate does not recognize if charge is started via API

### DISCUSSION THREAD

**Discussion by pkuehnel** [tobiasehlert/teslamateapi#138] [General]
I created a service to charge my Tesla based on sun on my pv roof. I am experiencing the problem, that when charging is stopped by my service TeslaMate tries to let the car sleep. Sometimes when I start charging again TeslaMate does not recognize that charging has started and TeslaMateAPI does not update its values. Is there any way to tell TeslaMate to abort "trying to sleep"?

**Comment by tobiasehlert** [comment]
Hello @pkuehnel,

The reason for TeslaMate not to react on the start if charge is that the API doesn't instruct TeslaMate to start logging. TeslaMate will receive an update from the Tesla API that the car is charging through the subscription it has.

If you want to trigger TeslaMate to start logging and not sleep, there is the `PUT /api/v1/cars/:CarID/logging/resume` endpoint that you can try to make a request to, when you are instructing your car to start logging :)

Please give it a try and get back to me with a result :)

Regards,
Tobias

**Reply by pkuehnel** [reply]
> TeslaMate will receive an update from the Tesla API that the car is charging through the subscription it has.

Yes but sometimes that does not do what it should.

I tried the `PUT` request, works fine, thank you.
